### PR TITLE
Fix up some internal dependences.

### PR DESCRIPTION
**Pull request by mhammond**
Takes the easy bits of #6129

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due diligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

File: components/support/viaduct-reqwest/src/lib.rs [hunk 1]
```diff
@@ -114,12 +114,3 @@ pub fn use_reqwest_backend() {
 pub extern "C" fn viaduct_use_reqwest_backend() {
     use_reqwest_backend();
 }
-
-/// A dummy symbol we include so that we can detect whether or not the reqwest
-/// backend got compiled in.
-#[no_mangle]
-pub extern "C" fn viaduct_detect_reqwest_backend() {
```
Comment (mhammond) [inline components/support/viaduct-reqwest/src/lib.rs:121]: this was from Tarik - @bendk, do you know whether this is actually safe to remove when considering desktop?

Comment (bendk) [reply components/support/viaduct-reqwest/src/lib.rs:121]: I think so?  I don't see that when I search in searchfox.

Comment (mhammond) [reply components/support/viaduct-reqwest/src/lib.rs:121]: Thanks - I hope you mean *does* look safe to remove, so I'll mark this as ready

Comment (linabutler) [reply components/support/viaduct-reqwest/src/lib.rs:121]: I went spelunking, and it looks like Thom added this way back in #937, as a check to ensure that we never included the Reqwest backend in our (then-Android only) Megazords—and which Ed removed in #3191.

**Review by linabutler — APPROVED** [review]

**Pull request merged**